### PR TITLE
Write feature test to replicate scenario with where user modifies JudgeAssignTask in multiple tabs

### PR DESCRIPTION
**Pull request by eileen-nava**
Resolves [Multiple Judge/Attorney Tasks](https://github.com/department-of-veterans-affairs/dsva-vacols/issues/212#).

### Description

The purpose of this PR was to replicate the scenario where a user creates multiple Judge and Attorney tasks by doing the same actions in two different tabs, as described in [this comment from Yoom](https://github.com/department-of-veterans-affairs/dsva-vacols/issues/212#issuecomment-889401070).  The user's actions cause a `JudgeAssignTask` to change status from `cancelled` to `completed`, which is an invalid flow.

(The below section is no longer relevant because JC and Yoom answered my question, but I am including it for posterity's edification.)
~~I was unable to completely replicate the scenario, because I couldn't figure out how to select "Assign an attorney" when there were two dropdowns on the page with this option.  Here is the error that the test currently throws.~~

```
Failures:

  1) duplicate JudgeAssignTask investigation Judge reassigns JudgeAssignTask in first tab and complete task in second tab Caseflow creates multiple JudgeDecisionReview and JudgeAssign tasks
     Failure/Error: container.find(selector, **keyword_args)
     
     Capybara::Ambiguous:
       Ambiguous match, found 2 elements matching visible css ".cf-select__control"
     # ./spec/support/feature_helper.rb:120:in `find_dropdown'
     # ./spec/support/feature_helper.rb:24:in `click_dropdown'
     # ./spec/fixes/investigate_duplicate_judge_tasks_spec.rb:61:in `block (4 levels) in <top (required)>'
     # ./spec/fixes/investigate_duplicate_judge_tasks_spec.rb:46:in `block (3 levels) in <top (required)>'

Finished in 37.65 seconds (files took 9.74 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/fixes/investigate_duplicate_judge_tasks_spec.rb:24 # duplicate JudgeAssignTask investigation Judge reassigns JudgeAssignTask in first tab and complete task in second tab Caseflow creates multiple JudgeDecisionReview and JudgeAssign tasks
```

~~I tried to open the test in a local browser and then select the "Assign an attorney" option based on its css selector, which is how I had selected the dropdown itself.  However, the dev tools did not display the css selectors for the task action options.... I am unsure why.~~

~~Guidance on how to select the task option would be appreciated, please!~~

### Acceptance Criteria
- [ ] Test replicates the scenario where the user opens multiple tabs and causes a `JudgeAssignTask` to change status from `cancelled` to `completed`

### Testing Plan
1. Run the test

### CODE REVIEW

File: spec/fixes/investigate_duplicate_judge_tasks_spec.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        expect(page).to have_content(COPY::ASSIGN_TASK_SUCCESS_MESSAGE, attorney_user.full_name)
+
+        visit "/queue/appeals/#{appeal.uuid}"
+        # Complete JudgeAssignTask to create an invalid number of AttorneyTasks
+        # TODO - figure out how to select an option from the second dropdown
+        # below correctly selects the second task action dropdown
+        find("#currently-active-tasks > div:nth-child(4) > table > tbody > tr:nth-child(3) > td.taskContainerStyling.taskActionsContainerStyling > div > div > div > div > div > div").click
+        # below - could not successfully select the "assign to attorney" dropdown because there were two dropdowns with this option
+        click_dropdown(text: "Assign to attorney")
```
Comment (jcq) [inline spec/fixes/investigate_duplicate_judge_tasks_spec.rb:61]: You should be able to scope it by specifying the `container` (second argument, after the options hash):
```rb
assign_task_row = find("#currently-active-tasks > div:nth-child(4) > table > tbody > tr:nth-child(3)")
        click_dropdown({text: "Assign to attorney"}, assign_task_row)
```

Comment (yoomlam) [reply spec/fixes/investigate_duplicate_judge_tasks_spec.rb:61]: I like `assign_task_row = find("table[summary='layout table'] > tbody > tr:nth-child(3)")` a bit better.

File: spec/fixes/investigate_duplicate_judge_tasks_spec.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        click_dropdown(prompt: "Select an action", text: "Assign to attorney")
+        click_dropdown(prompt: "Select a user", text: "Other")
+        click_dropdown(prompt: "Select a user", text: attorney_user_second.full_name)
+        fill_in "taskInstructions", with: "mimic incorrect flow"
+        click_on "Submit"
+        expect(page).to have_content(appeal.veteran.first_name, wait: 30)
+        appeal.reload.treee
+
+        # app currently allows the invalid flow of a JudgeAssignTask going from cancelled to completed
+        expect(Task.find(first_judge_assign_task_id).status).to eq("completed")
```
Comment (yoomlam) [inline spec/fixes/investigate_duplicate_judge_tasks_spec.rb:78]:
```suggestion
        # BUG: app currently allows the invalid flow of a JudgeAssignTask going from cancelled to completed
        expect(Task.find(first_judge_assign_task_id).status).to eq("completed")
```

File: spec/fixes/investigate_duplicate_judge_tasks_spec.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    let(:uuid) { appeal.uuid }
+
+    scenario "Caseflow creates multiple JudgeDecisionReview and JudgeAssign tasks" do
+      # open a window and visit case details page, window A
+      visit "/queue/appeals/#{uuid}"
+      expect(page).to have_content(appeal.veteran.first_name, wait: 30)
+      appeal.reload.treee
+
+      # open a second window and visit case details page, window B
+      second_window = open_new_window
```
Comment (pbradin) [inline spec/fixes/investigate_duplicate_judge_tasks_spec.rb:37]: This is so cool and great to see! Very well-described too.

**Review by pbradin — APPROVED** [review]
Nice job Eileen!

**Comment by eileen-nava** [comment]
Update: I decided to merge this branch into `eileen/2075-multiple-judge-tasks` and then merge `eileen/2075-multiple-judge-tasks` into master.  It seems simpler than having two separate PRs.

**Pull request merged**